atlp-rwanda/commanders-rn-medica
# Feature:  fill up profile and sign in and signup implementation

#### What does this PR do?
This PR implements the functionality for users to fill out their profile, sign in, and sign up.

#### Description of Task to be completed?
- Create a user profile screen where users can input their personal information including full name, nickname, date of birth, phone number, and gender.
- Implement functionality to upload and display a profile picture.
- Validate the input fields to ensure all required fields are completed and the phone number is in the correct format.
- Ensure the profile data is saved to the Supabase database.
- Handle user authentication for signing in and signing up.

#### How should this be manually tested?
1. Clone the repository and checkout to this branch.
2. Run `npm install` to install the dependencies.
3. Run the application on a simulator or physical device.
4. Navigate to the profile screen.
5. Fill out the profile form and upload a profile picture.
6. Click the "Continue" button and verify that the profile data is saved and a success message is displayed.
7. Test the sign-in and sign-up functionalities to ensure they are working correctly.

#### Any background context you want to provide?
This feature requires setting up a Supabase project and enabling Row Level Security policies for the `profile` table and `files` storage. Ensure the Supabase client is properly authenticated before making any insert or upload requests.

#### Questions:
- Do we need any additional validations for the profile form fields?
- Should we implement any additional user roles or permissions?
